### ISSUE TITLE
Error running force-tile-extent: ls: cannot access 'X*/*.tif'

### ISSUE THREAD

**Issue by thielfab** [davidfrantz/force#138]
Command:
`$ force-tile-extent ~/temp/ger_diss.shp ~/edc/level2 ~/temp/ger_tile_list.txt`

Output:
```                                 
ls: cannot access 'X*/*.tif': No such file or directory                                                 
dirname: missing operand                                                                                
Try 'dirname --help' for more information.                                                              
                                                                                                        
Suggested Processing extent:                                                                            
X_TILE_RANGE = 9999 -999                                                                                
Y_TILE_RANGE = 9999 -999                                                                                
                                                                                                        
Processing extent is not square.                                                                        
Suggest to use the tile allow-list:                                                                     
FILE_TILE = ~/temp/ger_tile_list.txt
```                                                       

~~Maybe something wrong in https://github.com/davidfrantz/force/blob/master/bash/force-tile-extent.sh#L73 ?~~

Calling `ls -d X*/*.tif` directly in datacube dir returns
`bash: /usr/bin/ls: Argument list too long`
or
`bash: /opt/singularity-3.8.2/bin/singularity: Argument list too long` (running command from within container)

Restricting output by using more specific queue like `ls -d X0065_Y005*/*.tif` works. So it probably is an issue with my systems argument length (see e.g.  [here](https://unix.stackexchange.com/questions/38955/argument-list-too-long-for-ls)).
Hence, this issue can be closed I guess...

- FORCE version: docker - davidfrantz/force:3.7.3-fix
- Container run using Singularity-3.8.2
- Host system: Ubuntu Server 21.10 (Kernel 5.13.0-16-generic)

**Comment by davidfrantz** [comment]
Hi Fabian,

damn... I really hoped this would not be a problem in practice.

``ls`` has this deficiency of refusing to work on long lists... Thus, this is not limited to your system. You probably have a large study area though.

Using ``find`` might be needed here... I hope I can look into this soon.
If you have a working substitute for ``ls -d X*/*.tif``, e.g. based on ``find``, this might speed things up for me.

Cheers,
David

**Comment by ernstste** [comment]
I think the issue here is that the script uses the base directory of the input vector file to create a data cube structure (to find the mask files that were created in this structure).
If the input vector happens to be located in your data cube, the `ls` command returns an amount of arguments that is too high to be handled by the shell.

**Comment by davidfrantz** [comment]
ah okay, yes this makes much sense :astonished:

> A QA engineer walks into a bar. Order a beer. Orders 0 beers. Orders 9999999999999 beers. Orders a lizard. Orders -1 beers. Orders a dfpsnfipned.
> First real customer walks in and asks where the bathroom is. The bar bursts into flames, killing everyone.

Then I see two solutions for developing a fix
- abort when there already is a datacube
- use a subfolder that is generated by the tool

**Comment by davidfrantz** [comment]
hmm, once inspecting more closely, I assume that the call to ``force-cube`` is the problem.

The subfolder logic was already implemented.

It still follows the older syntax, not the new one with the Linux-Style options.

This should be fixed with https://github.com/davidfrantz/force/commit/76d95c15e144739a67a81ed9f7fb36ec5e0085a0.

@thielfab, can you verify please (``develop`` for now).

**Comment by davidfrantz** [comment]
``force-tile-extent`` is still on my *Todo*-List for switching over to the new syntax, btw.

**Comment by thielfab** [comment]
> 
> 
> hmm, once inspecting more closely, I assume that the call to `force-cube` is the problem.
> 
> The subfolder logic was already implemented.
> 
> It still follows the older syntax, not the new one with the Linux-Style options.
> 
> This should be fixed with [76d95c1](https://github.com/davidfrantz/force/commit/76d95c15e144739a67a81ed9f7fb36ec5e0085a0).
> 
> @thielfab, can you verify please (`develop` for now).

Yes, seems to work without any issue now 👍. Output looks right and intermediate files are also removed again.

**Comment by davidfrantz** [comment]
Great!
